unity-renderer/Assets/Rendering/Culling/CullingController.cs

```diff
@@ -143,52 +143,42 @@ internal IEnumerator ProcessProfile(CullingControllerProfile profile)
+        }
 
+        internal IEnumerator ProcessProfileWithEnabledCulling(CullingControllerProfile profile, IEnumerable<Renderer> renderers)
+        {
+            Vector3 playerPosition = CommonScriptableObjects.playerUnityPosition;
+            float currentStartTime = Time.realtimeSinceStartup;
             foreach (Renderer r in renderers)
             {
-                if (timeBudgetCount > settings.maxTimeBudget)
-                {
-                    timeBudgetCount = 0;
-                    yield return null;
-                }
-
                 if (r == null)
                     continue;
 
-                bool rendererIsInIgnoreLayer = ((1 << r.gameObject.layer) & settings.ignoredLayersMask) != 0;
-
-                if (rendererIsInIgnoreLayer)
+                if (Time.realtimeSinceStartup - currentStartTime >= settings.maxTimeBudget)
                 {
-                    SetCullingForRenderer(r, true, true);
-                    continue;
+                    yield return null;
+                    playerPosition = CommonScriptableObjects.playerUnityPosition;
+                    currentStartTime = Time.realtimeSinceStartup;
                 }
 
-                float startTime = Time.realtimeSinceStartup;
-
-                //NOTE(Brian): Need to retrieve positions every frame to take into account
-                //             world repositioning.
-                Vector3 playerPosition = CommonScriptableObjects.playerUnityPosition;
-
                 Bounds bounds = MeshesInfoUtils.GetSafeBounds(r.bounds, r.transform.position);
-
                 Vector3 boundingPoint = bounds.ClosestPoint(playerPosition);
+
                 float distance = Vector3.Distance(playerPosition, boundingPoint);
                 float boundsSize = bounds.size.magnitude;
                 float viewportSize = (boundsSize / distance) * Mathf.Rad2Deg;
 
                 float shadowTexelSize = ComputeShadowMapTexelSize(boundsSize, urpAsset.shadowDistance, urpAsset.mainLightShadowmapResolution);
 
                 bool shouldBeVisible =
-                    // all objects are visible if culling is off
-                    !settings.enableObjectCulling
-                    // or if the player is inside the bounding box of the object
-                    || bounds.Contains(playerPosition)
-                    // or if the player distance is below the threshold
-                    || distance < profile.visibleDistanceThreshold
-                    // at last, we perform the expensive queries of emmisiveness and opaque conditions
-                    // these are the last conditions because IsEmissive and IsOpaque perform expensive lookups
-                    || viewportSize > profile.emissiveSizeThreshold && IsEmissive(r)
-                    || viewportSize > profile.opaqueSizeThreshold && IsOpaque(r)
+                    bounds.Contains(playerPosition) ||
+                    distance < profile.visibleDistanceThreshold ||
```

Comment:
Prefer this as first option to prevent executing the .Contains if possible, short circuit for performance. You may also all back the comments for the future maintainers

Reply:
I will be changing the order and getting back the comments for the last two lines, the lines about distance and bounds seem 100% understandable from the code and in my opinion, bring no value.

Reply:
It is clear what the line in the void does. Simply a condition, but overall we are doing a poor job commenting the code for other humans.. it is bad that we are removing precious bits of information for other maintainers